### PR TITLE
Refactoring to use default keyword, and other minor changes, for Source/Foundation

### PR DESCRIPTION
**Pull request by jonesmz**
Feel free to ask for changes.

### CODE REVIEW

File: Source/Foundation/bsfUtility/Reflection/BsIReflectable.cpp [hunk 1]
```diff
@@ -35,10 +35,9 @@ namespace bs
 	RTTITypeBase* IReflectable::_getRTTIfromTypeId(UINT32 rttiTypeId)
 	{
 		Stack<RTTITypeBase*> todo;
-		Vector<RTTITypeBase*>& rootClasses = getDerivedClasses();
 
-		for(auto iter = rootClasses.begin(); iter != rootClasses.end(); ++iter)
-			todo.push(*iter);
+		for(const auto & item : getDerivedClasses())
```
Comment (BearishSun) [inline Source/Foundation/bsfUtility/Reflection/BsIReflectable.cpp]: I prefer to keep the & next to the type, like so `const auto&`. I'm in the process of merging right now so I'll modify it on my end.

File: Source/Foundation/bsfUtility/Serialization/BsSerializedObject.h [hunk 2]
```diff
@@ -38,12 +38,10 @@ namespace bs
...
 	{
-		SerializedEntry()
-			:fieldId(0), serialized(nullptr)
-		{ }
+		SerializedEntry() = default;
 
-		UINT32 fieldId;
-		SPtr<SerializedInstance> serialized;
+		UINT32 fieldId = 0;
+		SPtr<SerializedInstance> serialized{nullptr};
```
Comment (BearishSun) [inline Source/Foundation/bsfUtility/Serialization/BsSerializedObject.h]: This one is redundant I believe, shared pointers are null by default.

**Comment by jonesmz** [comment]
Thanks for the info on the reference token. I'll try to keep that in mind.

Have you considered using something like clang-format, so automate formatting decisions like this?

Sure, I think you're right on the nullptr. I was kind of doing most of these changes mechanically, not really strongly thinking, while some other stuff was chugging on another computer. I probably did some other silly things as well.

**Comment by BearishSun** [comment]
That's a good suggestion, I added a new issue for it https://github.com/GameFoundry/bsf/issues/55.

**Pull request closed without merging**